x-pack/legacy/plugins/maps/common/descriptor_types.d.ts

```diff
@@ -119,3 +118,36 @@ export type VectorLayerDescriptor = LayerDescriptor & {
   joins?: JoinDescriptor[];
   style?: unknown;
 };
+
+export type RangeFieldMeta = {
+  min: number;
+  max: number;
+  delta: number;
+  isMinOutsideStdRange?: boolean;
+  isMaxOutsideStdRange?: boolean;
+};
+
+export type CategoryMeta = {
+  key: string;
+  count: number;
+};
+
+export type CategoryFieldMeta = {
+  categories: CategoryMeta[];
+};
+
+export type GeometryTypes = {
+  isPointsOnly: boolean;
+  isLinesOnly: boolean;
+  isPolygonsOnly: boolean;
+};
+
+export type StyleMetaValues = {
+  geometryTypes?: GeometryTypes;
+  fieldMeta: {
+    [key: string]: {
+      RANGE: RangeFieldMeta;
```

Comment:
Why is the key all caps? How about `range` instead of `RANGE`. and `categories` instead of `CATEGORIES `